pages/apim/3.x/kubernetes/apim-kubernetes-operator-user-guide-api-context.adoc

```diff
@@ -0,0 +1,174 @@
+
+Creating a secret to store the credentials:
+
+[source,yaml]
+----
+kubectl create secret generic management-context-credentials \
+  --from-literal=username=admin \
+  --from-literal=password=admin \
+  --namespace graviteeio
+----
+
+Defining an Api Context referencing the secret:
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: dev
+    organizationId: acme
+    auth:
+      secretRef:
+        name: management-context-credentials
+----
+
+NOTE: If no namespace has been specified for the secret reference, the management context resource namespace will be used to resolve the secret.
+
+NOTE: To target another environment on the same API instance, add another Management Context configured to do that.
+
+Although Kubernetes secrets should be the preferred way to store credentials, you can also add credentials inline in the Management Context definition:
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: dev
+    organizationId: acme
+    auth:
+      auth:
+        username: admin
+        password: admin
+----
+
+[,yaml]
+
+The example below uses a `bearerToken` to authenticate the requests. Note that the token must have been generated beforehand for the admin account:
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: staging
+    organizationId: acme
+    auth:
+      bearerToken: xxxx-yyyy-zzzz
+----
+
+Alternatively, here is how to use a Kubernetes secret to store the token:
+
+[source,yaml]
+----
+kubectl create secret generic management-context-credentials \
+  --from-literal=bearerToken=xxxx-yyyy-zzzz \
+  --namespace graviteeio
+----
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: staging
+    organizationId: acme
+    auth:
+      secretRef:
+        name: management-context-credentials
+----
+
+The example below uses inline credentials in the Management Context definition and a key value map to make the Api Definition manifest reusable and dynamic:
+
+[,yaml]
+----
+apiVersion: gravitee.io/v1alpha1
+kind: ApiContext
+metadata:
+  name: apim-example-context
+  namespace: graviteeio
+spec:
+  management:
+    baseUrl: https://gravitee-api.acme.com
+    environmentId: dev
+    organizationId: acme
+    auth:
+      auth:
+        username: admin
+        password: admin
+  values:
+    env: dev
+----
+
+{% raw %}
```

Comment:
What does this line do? It doesn't seem to have any impact on the HTML output.

Reply:
it avoids all templatings like ``{{.Values.env}}`` in the following yaml to be interpreted

Reply:
This section looks very odd when rendered as HTML. The text above mentions "The example below", but there seem to be two examples. Is that intentional? If it is, there needs to be more text explaining the examples.

Reply:
In the operator the Api Context is created only to be used with an ApiDefinition. Here we explain how to includes key values in the ApiContext. That's why there is two yaml. The user will need both examples to be able to do some templating. But I can rephrase to make it more clear